osu.Game.Tests/Visual/Online/TestSceneChatTextBox.cs

```diff
@@ -0,0 +1,123 @@
+
+        private OsuSpriteText commitText;
+        private OsuSpriteText searchText;
+        private ChatTextBar bar;
+
+        [SetUp]
+        public void SetUp()
+        {
+            Schedule(() =>
+            {
+                Child = new GridContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    RowDimensions = new[]
+                    {
+                        new Dimension(GridSizeMode.Absolute, 30),
+                        new Dimension(GridSizeMode.AutoSize),
+                    },
+                    Content = new[]
+                    {
+                        new Drawable[]
+                        {
+                            new GridContainer
+                            {
+                                RelativeSizeAxes = Axes.Both,
+                                ColumnDimensions = new[]
+                                {
+                                    new Dimension(),
+                                    new Dimension(),
+                                },
+                                Content = new[]
+                                {
+                                    new Drawable[]
+                                    {
+                                        commitText = new OsuSpriteText
+                                        {
+                                            Anchor = Anchor.TopCentre,
+                                            Origin = Anchor.TopCentre,
+                                            Font = OsuFont.Default.With(size: 20),
+                                        },
+                                        searchText = new OsuSpriteText
+                                        {
+                                            Anchor = Anchor.TopCentre,
+                                            Origin = Anchor.TopCentre,
+                                            Font = OsuFont.Default.With(size: 20),
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                        new Drawable[]
+                        {
+                            bar = new ChatTextBar
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Width = 0.99f,
+                            },
+                        },
+                    },
+                };
+
+                bar.OnChatMessageCommit += (sender, newText) =>
+                {
+                    commitText.Text = $"OnChatMessageCommit: {sender.Text}";
+                    commitText.FadeOutFromOne(1000, Easing.InQuint);
+                    sender.Text = string.Empty;
```

Comment:
Why is this text-clearing behaviour in the test rather than inside the chat box component? Is there some higher-level logic that is going to require the chat overlay as a whole to do this on its own?

Reply:
Given this event is supposed to be handled at the overlay level, I was naiively keeping it the same to how it currently is, and the test reflects it's intended usage. https://github.com/ppy/osu/blob/a2829b4fc7fa7a79e6e884fc9a381a9592eb7aad/osu.Game/Overlays/ChatOverlay.cs#L495-L508

I guess it's not really needed now so making the event `Action<string>` and having the text be cleared internally makes more sense now.